File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
    int l = 0;
      int r = 0;
      int n = s.size();
      int k = 3;
      unordered_map<int,int> mpp;
      int cnt = 0;  
      while(r < n)
      {
            mpp[s[r]]++;
            while(mpp.size() == 3)
            {
                mpp[s[l]]--;
                if(mpp[s[l]]==0)
                  mpp.erase(s[l]);
                l++;
                cnt = cnt + (n-r);
            }
            
            r++;
      }
      return cnt;  
    }
};

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {

     int l = 0;
     int r = 0;
     int distinct = 3;
     int count = 0;
     HashMap<Character, Integer> map = new HashMap<>();

     while(r!= s.length())
     {
        map.put(s.charAt(r), map.getOrDefault(s.charAt(r),0) + 1);
        while(map.size() == 3)
        {
            map.put(s.charAt(l), map.get(s.charAt(l)) - 1);

            if (map.get(s.charAt(l)) == 0) {
                map.remove(s.charAt(l));
            }
            l++;
            count = count + (s.length()-r);
        }
        r++;

     }
     return count;   
    }
}